### ISSUE TITLE
[DFC Orders] Link Products on external platforms

### ISSUE THREAD

**Issue by lin-d-hop** [openfoodfoundation/openfoodnetwork#11946]
*Funded Feature. Please track ALL ASSOCIATED WORK under the associated tracking code: #11678 DFC Orders*

## Description
In order to link between variants in OFN and variants in another platform we need to add a field that can serve as a link between the platforms.

Variants in OFN can have multiple links to variants in other platforms and vice versa eg retail and wholesale versions.
Thus we need to ensure the ability to have a many to many mapping.

## Acceptance Criteria
- New table that joins a variant to a URL. The URL is the product in another platform.
- No UI for this at this stage so no translations

**Comment by mkllnk** [comment]
Implementation detail: we need one new table referencing Spree::Variant and a DFC URI. One variant has many semantic URIs. And the same semantic URI can appear multiple times in the table and reference many variants.

Thoughts:

- The URIs are not the id of the variant because every variant has only one id.
- A retail variant can link to bulk variants with different quantities. We enable transformations here.

My best idea is to call the new table `semantic_links` which is very open for extension later and doesn't imply a specific use for back-ordering or source of truth.

**Comment by mkllnk** [comment]
@RaggedStaff We talked about linking to a DFC SuppliedProduct on another platform but I'm just wondering if we want to link directly to an Offer or CatalogItem. In the end, we want to place an Order which references offers. So linking to an offer would be the easiest. But if offers are replaced over time then we may want to query the SuppliedProduct and hope that the endpoint gives us a list of catalog items with a list of offers so that we can order. What do you think?

**Comment by lin-d-hop** [comment]
@mkllnk does this still need more feedback? Or do you  have enough to continue?

**Comment by mkllnk** [comment]
Yes, I updated the data model to add a quantity. Now we need to build the UI to make this useful. I don't think I need to get my current PR merged. It's only one simple commit and I can add that to the UI issue.
